ethcore/src/engines/authority_round.rs

```diff
@@ -241,8 +240,9 @@ impl Engine for AuthorityRound {
 		]
 	}
 
-	fn schedule(&self, _env_info: &EnvInfo) -> Schedule {
-		Schedule::new_post_eip150(usize::max_value(), true, true, true)
+	fn schedule(&self, block_number: BlockNumber) -> Schedule {
```

Comment:
There's a number of places where Schedule is created from the block number number. This PR introduces quite a few more. Creating a fake `EnvInfo` struct seems wasteful. And furthermore since all of the data except for the block number is unreliable it may introduce subtle bugs if any future implementation will have to rely on any other field.